internal/server/data/migrations.go

```diff
@@ -1003,7 +992,22 @@ ALTER TABLE ONLY user_public_keys
 CREATE UNIQUE INDEX IF NOT EXISTS idx_user_public_keys_user_fingerprint ON user_public_keys
     USING btree (fingerprint) WHERE (deleted_at IS NULL);
 `
-			// TODO: is this the right index for joining to the user table?
+			_, err := db.Exec(stmt)
+			return err
+		},
+	}
+}
+
+func addUserSSHLoginName() *migrator.Migration {
```

Comment:
As you mentioned in the description of this PR, I would personally rather see existing users get assigned SSH usernames in this migration even though it is heavy. It means we don't have to add checks for empty SSH usernames at other places in ways that we have to clean up in the future.

Reply:
I think our current approach to migrations won't really scale. At some point we're going to need to migrate data in batches separate from our deployment. I'm not sure exactly when that will be.

It's probably ok to do this migration all at deployment time. I'll check the number of users we have right now in the DB to try and get some idea of how safe this is.

Reply:
Ok, I added a commit that does the generation of `SSHLoginName` in the migration. There's a couple unfortunate things, but overall it is simpler and safer than having to do it later. Eventually we will need to start migrating data separately from deploys.